src/lib/providers/CurrencyProvider.tsx

```diff
@@ -26,18 +26,30 @@ const CurrencyProvider: React.FC<CurrencyProviderType> = ({ children }) => {
 		getLootPriceUsd();
 	});
 
-	// Poll CoinGecko every 30 seconds
+	// Poll CoinGecko every 30 seconds, 5 seconds if call fails and set value to $0 (wont be rendered in this case)
 	const getWildPriceUsd = async () => {
-		const price = await wildTokenPrice();
-		setWildPriceUsd(price || 0);
-		await new Promise((resolve) => setTimeout(resolve, 30000));
+		try {
+			const price = await wildTokenPrice();
+			setWildPriceUsd(price);
+			await new Promise((resolve) => setTimeout(resolve, 30000));
+		} catch (e) {
+			setWildPriceUsd(0);
```

Comment:
No need to reset the price to zero. Just leave it as the value it was previously (which will either be undefined or a valid amount)